File: python/unidad_02/problema_2_2.py
'''
El programa, dado como datos dos valores enteros, determina si los
mismos satisfacen una expresión.
'''

def calcular_expresion(numero_01, numero_02):

    return numero_01 ** 3 + numero_02 ** 4 - 2 * numero_01 ** 2
    # Expresión que contempla ambos números de entrada

def main():

    try:

        numero_01 = int(input("1er número: "))

        numero_02 = int(input("2do número: "))

    except ValueError as e:

        print("Error: por favor, ingresa un valor válido.")
        return
    
    resultado = calcular_expresion(numero_01, numero_02)

    condicion = resultado < 680
    # Valor booleano

    mensaje = "cumplen" if condicion else "no cumplen"

    simbolo = '<' if condicion else '>'

    print(f"P = {numero_01} y Q = {numero_02} {mensaje} con la condición")

    print(f"P ** 3 + Q ** 4 - 2 * P ** 2 < 680")

    print(f"{resultado:.0f} {simbolo} 680")

if __name__ == '__main__':

    main()

File: java/unidad_02/Problema_2_2/src/Problema_2_2.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dado como datos dos valores enteros, determina si los
 * mismo satisfacen una expresión.
 *
 * @author Angel P.
 * 
 */

public class Problema_2_2 {

    public static int calcularExpresion(int numero01, int numero02) {

        return (int) Math.pow(numero01, 3) + (int) Math.pow(numero02, 4) - 2 * (int) Math.pow(numero01, 2);
        // Expresión que contempla ambos números de entrada
    }

    public static void main(String[] args) {

        int numero01, numero02;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.println("1er número: ");
            numero01 = scanner.nextInt();

            System.out.println("2do número: ");
            numero02 = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println("Error: por favor, ingresa números enteros.");
            return;
        }
        int resultado = calcularExpresion(numero01, numero02);

        boolean condicion = resultado < 680;

        String mensaje = (condicion) ? "cumplen" : "no cumplen";

        String simbolo = (condicion) ? "<" : ">";

        System.out.printf("P = %d y Q = %d %s con la condición %n", numero01, numero02, mensaje);

        System.out.println("P ** 3 + Q ** 4 - 2 * P ** 2 < 680");

        System.out.printf("%d %s 680", resultado, simbolo);
    }
}